include/h2o.h

```diff
@@ -467,6 +467,16 @@ enum {
 };
 
 /**
+ * holds various data related to the context
+ */
+typedef struct st_h2o_context_storage_item_t {
+    void (*dispose)(struct st_h2o_context_storage_item_t *self, h2o_context_t *ctx);
```

Comment:
Also, please pass-in the `data` pointer as the first argument instead of `h2o_context_storage_item_t`.

That's the design generally found in H2O (and in C, IIRC).